perf/src/packet.rs

```diff
@@ -81,37 +81,39 @@ impl PacketBatch {
     }
 }
 
-pub fn to_packet_batches<T: Serialize>(xs: &[T], chunks: usize) -> Vec<PacketBatch> {
-    xs.chunks(chunks)
-        .map(|x| {
-            let mut batch = PacketBatch::with_capacity(x.len());
-            batch.packets.resize(x.len(), Packet::default());
-            for (i, packet) in x.iter().zip(batch.packets.iter_mut()) {
-                Packet::populate_packet(packet, None, i).expect("serialize request");
+pub fn to_packet_batches<T: Serialize>(data: &[T], chunk_size: usize) -> Vec<PacketBatch> {
+    data.chunks(chunk_size)
+        .map(|data_chunk| {
```

Comment:
`data_chunk` sounds like a sub-unit of a single packet's data. Maybe `batch_data` or `batch_items` would be clearer